python/lsst/daf/butler/datastores/chainedDatastore.py

```diff
@@ -872,18 +905,97 @@ def retrieveArtifacts(
 
         # Now do the transfer.
         targets: list[ResourcePath] = []
+        merged_artifacts_to_ref_id: dict[ResourcePath, list[DatasetId]] = {}
+        merged_artifacts_to_info: dict[ResourcePath, StoredFileInfo] = {}
         for number, datastore_refs in grouped_by_datastore.items():
-            targets.extend(
-                self.datastores[number].retrieveArtifacts(
-                    datastore_refs,
-                    destination,
-                    transfer=transfer,
-                    preserve_path=preserve_path,
-                    overwrite=overwrite,
-                )
+            retrieved, artifacts_to_ref_id, artifacts_to_info = self.datastores[number].retrieveArtifacts(
+                datastore_refs,
+                destination,
+                transfer=transfer,
+                preserve_path=preserve_path,
+                overwrite=overwrite,
+                write_index=False,  # Disable index writing regardless.
+                add_prefix=add_prefix,
+            )
+            targets.extend(retrieved)
+            merged_artifacts_to_ref_id.update(artifacts_to_ref_id)
+            merged_artifacts_to_info.update(artifacts_to_info)
```

Comment:
Could it happen that the same artifact path corresponds to different sets of uuids/records in different datastores?

Reply:
I don't think so. This is only retrieving datasets from the second datastore that it didn't fetch from the first datastore. The UUIDs are defined at the registry level.